.github/workflows/build.yaml

```diff
@@ -70,19 +70,34 @@ jobs:
           MACOSX_DEPLOYMENT_TARGET: 10.14
         run: |
           cargo build --verbose --release
-          mv target/release/eslauncher2 eslauncher2-x86_64-apple-darwin
+          chmod 755 target/release/eslauncher2
+          ls -al target/release
+      - name: Package binary
+        run: |
+          cargo install cargo-bundle
+          cargo bundle --release
+          ls -al target/release/bundle/osx/ESLauncher2.app/Contents/MacOS
       - name: Strip binary
-        run: strip eslauncher2-x86_64-apple-darwin
-      - name: Archive binary
-        uses: actions/upload-artifact@v1
+        run: |
+          strip target/release/bundle/osx/ESLauncher2.app/Contents/MacOS/eslauncher2
+          chmod 755 target/release/bundle/osx/ESLauncher2.app/Contents/MacOS/eslauncher2
+          ls -al target/release/bundle/osx/ESLauncher2.app/Contents/MacOS
+      - name: Package ourselves to prevent execution flag creep
+        run: |
+          cd target/release/bundle/osx
+          zip -r ESLauncher2.app.zip ESLauncher2.app
+          rm -r ESLauncher2.app
+          ls -al
```

Comment:
should get rid of this `ls` before merge. 

You can also ditch the `rm`, `upload-artifact` v2 does accept single files as `path`.

Reply:
Both removed. rm was a leftover from earlier, where I ended up with both the zip file and the app-subdirectory. But since I now specify the exact file, it should work.